src/app/reviewer/[orgSlug]/study/[studyId]/review/study-review-buttons.tsx

```diff
@@ -35,6 +37,42 @@ export const StudyReviewButtons: FC<{ study: SelectedStudy }> = ({ study }) => {
         onSuccess: () => router.push(backPath),
     })
 
+    const [isApproveModalOpen, { open: openApproveModal, close: closeApproveModal }] = useDisclosure(false)
+
+    // Show the confirmation modal again only if 48 hours have passed since the last action.
+    const HOURS_MS = 48 * 60 * 60 * 1000 // 48 hours
+    const ACTION_TIMESTAMP_KEY = 'studyApprovalActionAt'
+
+    // Persist a timestamp in localStorage whenever the user either approves the study or chooses to generate a new key from the modal.
+    const saveActionTimestamp = () => {
+        try {
+            localStorage.setItem(ACTION_TIMESTAMP_KEY, Date.now().toString())
+        } catch (error) {
+            console.error('Failed to write to localStorage', error)
+        }
+    }
+
+    // When the "Approve" button is clicked:
+    // • If a modal action (approve or generate key) occurred within the past 48 hours, skip the modal and approve immediately.
+    // • Otherwise, display the confirmation modal.
+    const showConfirmationModal = () => {
+        if (isPending || isSuccess) {
+            return
+        }
+
+        try {
+            const lastAction = localStorage.getItem(ACTION_TIMESTAMP_KEY)
+            if (lastAction && Date.now() - parseInt(lastAction, 10) < HOURS_MS) {
+                updateStudy('APPROVED')
+                return
+            }
+        } catch (error) {
+            console.error('Failed to read from localStorage', error)
+        }
+
+        openApproveModal()
+    }
+
```

Comment:
lets make this be an external hook: `useShouldPromptForKeyConfirmation` or something, so we can update and test the logic independently